pkg/kubelet/images/image_manager.go

```diff
@@ -29,27 +29,29 @@ import (
 
 // imageManager provides the functionalities for image pulling.
 type imageManager struct {
-	recorder record.EventRecorder
-	runtime  kubecontainer.Runtime
-	backOff  *flowcontrol.Backoff
+	recorder     record.EventRecorder
+	imageService kubecontainer.ImageService
+	backOff      *flowcontrol.Backoff
 	// It will check the presence of the image, and report the 'image pulling', image pulled' events correspondingly.
 	puller imagePuller
 }
 
 var _ ImageManager = &imageManager{}
 
-func NewImageManager(recorder record.EventRecorder, runtime kubecontainer.Runtime, imageBackOff *flowcontrol.Backoff, serialized bool) ImageManager {
+func NewImageManager(recorder record.EventRecorder, imageService kubecontainer.ImageService, imageBackOff *flowcontrol.Backoff, serialized bool, qps float32, burst int) ImageManager {
+	imageService = newThrottledPuller(imageService, qps, burst)
```

Comment:
From your recent change, I found that in fact the name `newThrottledPuller` is a little misleading, because this returns an image service not a puller.

Just learnt the decorator pattern recently, may be useful here https://github.com/smaxwellstewart/golang-decorator-pattern/blob/master/client/decorate.go.